src/js/components/SidebarComponent.jsx

```diff
@@ -0,0 +1,106 @@
+var Link = require("react-router").Link;
+var React = require("react/addons");
+
+var AppListLabelsFilterComponent =
+  require("../components/AppListLabelsFilterComponent");
+var AppListStatusFilterComponent =
+  require("../components/AppListStatusFilterComponent");
+
+var SidebarComponent = React.createClass({
+  displayName: "SidebarComponent",
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    groupId: React.PropTypes.string.isRequired,
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      filters: {
+        filterLabels: [],
+        filterStatus: []
+      }
+    };
+  },
+
+  updateFilterLabels: function (filterLabels) {
+    var state = this.state;
+
+    this.setState(Object.assign(state.filters, {
```

Comment:
Careful. You are modifying state directly this way. What you really want is
```
this.setState(Object.assign({}, state.filters, {
// ...
```
But I'd really much rather prefer using the immutability helpers. 
```
var filterLabels = React.addons.update(state.filterLabels, {$merge: filterLabels});
this.setState({filterLabels: filterLabels});
```
(haven't tested but it should be close)
https://facebook.github.io/react/docs/update.html